### ISSUE TITLE
Deutliche Lageabweichungen

### ISSUE THREAD

**Issue by hbruch** [mfdz/zhv-issues#16]
Für mehr als 3600 Haltestellen gehen wir von Lageabweichnungen größer 200 Meter aus.

Zu dieser Einschätzung gelangen wir aufgrund eines Matchings von OSM-Haltestellen mit den Haltestellen des zHV. OSM-Halte zeichnen sich in der Regel durch hohe Aktualität und Lagegenauigkeit aus, so dass wir die fehlerhafte Koordinatenangabe überwiegend beim zHV sehen. 

In seltenen Fällen dürften auch Falschzuordnungen (z.B. aufgrund fehlender Haltestellen in OSM oder Matching verlegter Haltestellen und daraus folgender Falsch-Zuordnungen) die Ursache sein, was aus unserer Sicht jedoch nur für einen sehr kleinen Teil zutrifft.

Nach Bundesland ergibt sich folgende Verteilung:
Land|Anzahl
-- | --
de:01|133
de:02|11
de:03|409
de:04|6
de:05|660
de:06|66
de:07|194
de:08|501
de:09|575
de:10|35
de:11|21
de:12|97
de:13|352
de:14|215
de:15|297
de:16|110

Details sind angehängter [Liste](https://github.com/mfdz/zhv-issues/files/6954333/issue_16.csv) zu entnehmen.

**Stand der Daten**
zHV: 09.07.2021
OSM: 16.07.2021

**Comment by hbruch** [comment]
In Bezug auf die Verantwortlichkeiten verteilen sich de Lageabweichungen wie folgt:


Verbund|Count(*)
-- | --
BEG|568
Hamburger Hochbahn|62
HannIT|121
NASA|408
NVBW|495
NVV|12
VBB|118
VBN|376
VMT|101
VMV|352
VRN|213
VRR|660
VVO|113
ZPS|35
rms|48

Regional sind die Vorkommen wie folgt verteilt:
<img width="323" alt="image" src="https://user-images.githubusercontent.com/2187389/128720888-cda9f358-7857-47f9-bedb-5970c5492ced.png">

**Comment by NVBWSeifert** [comment]
Hallo Holger,

kannst Du für Baden-Württemberg und uns (NVBW) die Liste der stark abweichenden Haltestellen BW-weit bereitstellen, damit wir dem konkret im Einzelfall nachgehen können?

**Comment by hbruch** [comment]
Hilft Euch bereits die oben angehängte Liste, wenn Ihr sie auf die "de:08"-Einträge filtert?

**Comment by NVBWSeifert** [comment]
> 
> 
> Hilft Euch bereits die oben angehängte Liste, wenn Ihr sie auf die "de:08"-Einträge filtert?

Ja, habe den Link übersehen gehabt, genau die Liste brauchen wir.
Wir machen uns dann mal dran, die Zahl zu verringern, wird aber länger dauern ;(

**Comment by MarcelVRS** [comment]
Unsere ca 50 im VRS werden wir sichten und korrigieren

**Comment by hbruch** [comment]
Der Datensatz vom 2.1.2023 enthält noch 2154 Steige mit Abweichungen größer 200m.

Gruppiert nach Verantwortlichen sind diese wie folgt verteilt:

Verbund | Anzahl
-- | --
BEG|554
Hamburger Hochbahn|47
HannIT|63
NASA|11
NVBW|206
NVV|1
VBB|82
VBN|357
VMT|10
VMV|64
VRN|92
VRR|571
VVO|53
ZPS|12
rms|31

**Comment by MarcelVRS** [comment]
Würde dann mal sagen, sieht sehr gut aus für den VRS 😉

**Comment by hbruch** [comment]
Mit Stand 4.3.2024 bestehen weiterhin > 2300 Abweichungen (ermittelt mittels [nvbw-osm-stop-comparison](https://github.com/mfdz/nvbw-osm-stop-comparison)). Hierbei sind Abweichungen > 400 Meter noch nicht berücksichtigt. 

Die häufigsten sehen wir in den folgenden Kreisen:
|  Kreis   |               Name               | Abweichend | Gesamtzahl | Prozent |
|----------|----------------------------------|------------|------------|---------|
| de:09374 | Landkreis Neustadt a.d. Waldnaab | 44         | 1267       | 3.47    |
| de:09475 | Landkreis Hof                    | 28         | 825        | 3.39    |
| de:05766 | Kreis Lippe                      | 80         | 3159       | 2.53    |
| de:09278 | Landkreis Straubing-Bogen        | 34         | 1346       | 2.53    |
| de:01061 | Kreis Steinburg                  | 29         | 1185       | 2.45    |
| de:09677 | Landkreis Main-Spessart          | 24         | 986        | 2.43    |
| de:03102 | Kreisfreie Stadt Salzgitter      | 12         | 499        | 2.4     |
| de:09476 | Landkreis Kronach                | 13         | 551        | 2.36    |
| de:05754 | Kreis Gütersloh                  | 53         | 2358       | 2.25    |
| de:03462 | Landkreis Wittmund               | 14         | 636        | 2.2     |

SQL-Statement zur Ermittlung:
```sql
SELECT 'de:'||v.ags Kreis, bez || ' ' || gen Name ,distant Abweichend, total Gesamtzahl, round(distant*100.0/total,2) Prozent
  FROM vg250_krs v
  JOIN (SELECT district, value distant
         FROM match_stats m
        WHERE m.key = 'MATCHED_THOUGH_DISTANT'
          AND run_id = (SELECT max(run_id) FROM match_meta_data )) d ON d.district='de:'|| v.AGS
  JOIN (SELECT district, sum(value) total
         FROM match_stats m
        WHERE m.key NOT IN ('NO_MATCH_AND_SEEMS_UNSERVED')
          AND run_id = (SELECT max(run_id) FROM match_meta_data )
        GROUP BY district) u ON u.district='de:'|| v.AGS
  WHERE v.GF=4
  ORDER by distant*100.0/total desc
  LIMIT 10;
```
Laut zHV nicht bediente Steige (welche in Realität aufgrund eventuell fehlender Fahrplandaten durchaus bedient sein können) haben wir nicht zur Gesamtzahl gezählt.

**Comment by CM-RMS** [comment]
Dazu habe ich eine Frage - Beispiel ist diese Haltestelle:
![image](https://github.com/mfdz/zhv-issues/assets/108124275/04863991-bf27-426d-9626-c0e19eb0c23e)
Im zHV sind diese Koordinaten:
![image](https://github.com/mfdz/zhv-issues/assets/108124275/3d31b703-4fb6-40d7-a63b-9980fafae786)
Die zHV Koordinaten aus der Liste weichen also nicht nur von OSM ab sondern auch von dem was im zHV angegeben ist. Von welchen Koordinaten reden wir jetzt?

**Comment by hbruch** [comment]
Die oben angehängte Liste datiert von August 2021. Ich hoffe sehr, dass einige Fehler mittlerweile behoben wurden.

Die aktuellen Abweichungen sind unter den kreisweisen Auswertungen unter https://gtfs.mfdz.de/delfi/ einsehbar.

Im [Abgleich-Report](https://gtfs.mfdz.de/delfi/region_de01002.html) der letzten Wochen zHV Stand 2024-03-18 sind die Koordinaten wie folgt angegeben:
<img width="914" alt="image" src="https://github.com/mfdz/zhv-issues/assets/2187389/f3e25dc4-cbce-443e-92fd-e2c61a7a38bb">
Analog im [ausgeleiteten CSV](https://gtfs.mfdz.de/delfi/region_de01002.csv):
<img width="702" alt="image" src="https://github.com/mfdz/zhv-issues/assets/2187389/fc64dea4-0501-45d8-bf43-ed9b2ac328b2">
Damit entsprechen sie, bis auf Rundungsabweichungen, denen des zHV Screenshots. 

Enthielte die CSV-Ausleitung des zHV wie in #11 die Angabe des letzen Änderungsdatums im zHV ließe sich sogar der mutmaßliche Zeitpunkt der Korrektur erkennen.
